internal/metric/mock.go

```diff
@@ -54,101 +48,83 @@ type (
 
 	Meter struct {
 		MeasurementBatches []Batch
-		// Observers contains also unregistered
-		// observers. Check the Dead field of the Observer to
-		// figure out its status.
-		Observers []*Observer
+		AsyncInstruments   []*Asynchronous
 	}
 
-	Kind int8
-
 	Measurement struct {
 		// Number needs to be aligned for 64-bit atomic operations.
 		Number     core.Number
-		Instrument *Instrument
+		Instrument apimetric.InstrumentImpl
 	}
 
-	observerResult struct {
-		instrument *Instrument
+	Instrument struct {
+		meter      *Meter
+		descriptor apimetric.Descriptor
 	}
 
-	int64ObserverResult struct {
-		result observerResult
-	}
+	Asynchronous struct {
+		Instrument
 
-	float64ObserverResult struct {
-		result observerResult
-	}
+		Dead bool
```

Comment:
You could drop it - it was used for observer unregistration, which you removed.

Reply:
Sorry about this. I'm happy to restore Unregister in another PR, but I don't think anyone else has mentioned it.